helpers/config.py

```diff
@@ -581,6 +581,59 @@ def validate_aws_credentials(self):
         )
         self.__dict['aws_credentials_valid'] = validation.validate_credentials()
 
+    def validate_passwords(self):
+
+        default_pattern = rf'^[{self.ALLOWED_PASSWORD_CHARACTERS}]{{8,}}$'
+
+        passwords = {
+            'PostgreSQL': {
+                'password': self.__dict['postgres_password'],
+                'pattern': default_pattern,
+            },
+            'PostgreSQL replication': {
+                'password': self.__dict['postgres_replication_password'],
+                'pattern': default_pattern,
+            },
+            'MongoDB root’s': {
+                'password': self.__dict['mongo_root_password'],
+                'pattern': default_pattern
+            },
+            'MongoDB user’s': {
+                'password': self.__dict['mongo_user_password'],
+                'pattern': default_pattern
+            },
+            'Redis': {
+                'password': self.__dict['redis_password'],
+                # redis password can be empty
+                'pattern': rf'^[{self.ALLOWED_PASSWORD_CHARACTERS}]{{8,}}|$'
+            },
+        }
+        errors = []
+        for label, config_ in passwords.items():
+            if not re.match(config_['pattern'], config_['password']):
+                errors.append(label)
+                CLI.colored_print(
+                    f'{label} password contains unsupported characters.',
+                    CLI.COLOR_WARNING
+                )
```

Comment:
Maybe rather something like this? But it's not really important...

```python3

def get_pattern(chars=8, allow_empty=False):
  pattern = f'[{self.ALLOWED_PASSWORD_CHARACTERS}]{{{chars},}}'
  if allow_empty:
    pattern += '|'
  return rf'^{pattern}$'

...

passwords = {
  ...
  'redis': {
    'label': 'Redis',
    'pattern': get_pattern(allow_empty=True)
  }
}
errors = []
for name, config_ in passwords.items():
  if not re.match(config_['pattern'], self.__dict[f'{name}_password']):
    errors.append(name)
    CLI.colored_print(
      f'{config_['label']} password contains unsupported characters.',
      CLI.COLOR_WARNING
    )
```

Reply:
Great idea ;-)